### DISCUSSION TITLE
tutorials/performance/vertex_animation/controlling_thousands_of_fish

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#56] [User-contributed notes]
# tutorials/performance/vertex_animation/controlling_thousands_of_fish

The problem with MeshInstance3D is that it is expensive to update their transform array. It is great for placing many static objects around the scene. But it is still difficult to move the objects ...

https://docs.godotengine.org/en/stable/tutorials/performance/vertex_animation/controlling_thousands_of_fish.html

**Comment by samkoesnadi** [comment]
Since this is using particles, it means the fishes will be gone after some "lifetime" duration right?

**Reply by Calinou** [reply]
Fishes will be gone after the lifetime duration, but if **Oneshot** is unchecked, new fish will reappear to replace the fish that disappeared.

There is currently no support for particles with infinite lifetime, but you could use a high lifetime and an equivalent **Preprocess** value so that fishes aren't replaced often. I probably wouldn't go above 300 though, as preprocessing several minutes of particles will cause stuttering when particles are first loaded in (since the engine needs to simulate particles up to that point before it can render a frame).

**Reply by samkoesnadi** [reply]
Understood... This is very good to know, especially to decide whether to use multimesh or particles approach. Big thanks!
